src/ProtocolStreamReader.php

```diff
@@ -30,33 +30,32 @@ public function __construct($input)
     {
         $this->input = $input;
         Loop\addReadStream($this->input, function() {
-            while (($c = fgetc($this->input)) !== false) {
-                $this->buffer .= $c;
-                switch ($this->parsingMode) {
-                    case ParsingMode::HEADERS:
-                        if ($this->buffer === "\r\n") {
-                            $this->parsingMode = ParsingMode::BODY;
-                            $this->contentLength = (int)$this->headers['Content-Length'];
-                            $this->buffer = '';
-                        } else if (substr($this->buffer, -2) === "\r\n") {
-                            $parts = explode(':', $this->buffer);
-                            $this->headers[$parts[0]] = trim($parts[1]);
-                            $this->buffer = '';
+            $c = fgetc($this->input);
```

Comment:
In my latest commit I added a `return` after a full message has been parsed. This should fix the problem with `fgetc` as the control is returned to the mainloop where `stream_select` is called again.

The real bummer is that `stream_select` on windows behaves erratically with STDIN.
Even if I set timeout to zero, it will always block without data on the stream. :/

If I understand correctly, there exists some very kludgy workarounds for this problem. NodeJS probably implemented them, while PHP did not. :/